Set up the doc better and add the missing secrets template (#11)
NVIDIA/nim-deploy: helm/README.md

Each NIM contains an AI model, application, or workflow. All files necessary to run the NIM are encapsulated in the container that is available on [NGC](https://ngc.nvidia.com/). The [NVIDIA API Catalog](https://build.nvidia.com) provides a sandbox to experiment with NIM APIs prior to container and model download.

## Setting up your helm values

All available helm values can be discoved by running the `helm` command after downloading the repo.

```bash
helm show values nim-llm/
```

> See the chart's [readme](nim-llm/README.md) for information about the options, including their default values. Pay particular attention to the image.repository and image.tag options if you do not want to deploy the default image for this chart..

Reviewer: ```suggestion
See the chart's [readme](nim-llm/README.md) for information about the options, including their default values. Pay particular attention to the image.repository and image.tag options if you do not want to deploy the default image for this chart.
```

To run llama3-8b-instruct using the NIM 1.0.0 release, you need to change very little and don't even truly need a values file. For an example, see [Launching a default NIM with minimal values](#Launching-a-default-NIM-with-minimal-values)

## Create a namespace

For the rest of this guide, we will use the namespace `nim`.

```bash
kubectl create namespace nim
```

## Launching a default NIM with minimal values

`llama3-8b-instruct` at the chart's appVersion (currently 1.0.0) can be launched in a default configuration while only setting the NGC API key and persistence in one line with no extra files. Using Persistent volumes is still recommended to ensure permissions are set correctly and the container runtime filesystem isn't filled by downloading models.

```bash
helm --namespace nim install my-nim nim-llm/ --set model.ngcAPIKey=$NGC_CLI_API_KEY --set persistence.enabled=true
```

## Using a custom values file

When deploying NIMs there are several values that can be customized to control factors such as scaling, metrics collection, resource use, and most general AI model selection.

> Here is an example using meta/llama-3-8b-instruct. Where you have a pre-existing secret for the NGC API key, using persistent value stores to maintain the model cache and a specific non-default image pull secret. This also disables the StatefulSet setup so that it deploys a Deployment object instead.

Reviewer: ```suggestion
The following example uses meta/llama-3-8b-instruct with an existing secret as the value of the NGC API key, sets the persistent value stores to maintain the model cache and a specific non-default image pull secret, and disables the StatefulSet setup so that it deploys a Deployment object instead.
```

> If specifying secrets like this, one cannot necessarily set the API key directly in the file or on the CLI, so the cluster operator should create the secrets first like:

Reviewer: ```suggestion
If you specify secrets as shown in this example, you cannot set the API key directly in the file or on the CLI. Instead, first create the secrets, as shown in the following example:
```

    nvidia.com/mig-4g.24gb: 1
```

### Launching NIM in Kubernetes with a values file

> A command like the one below will then use the latest chart version to install the version of NIM defined in the values file into the `nim` namespace in your Kubernetes cluster. Modify it as required.

Reviewer: ```suggestion
Then use a command like the following, which uses the latest chart version to install the version of NIM that is defined in the values file into the `nim` namespace in your Kubernetes cluster.
```

If you are operating on a fresh persistent volume or similar, you may have to wait a little while for the model to download. You can check the status of your deployment by running

```bash
kubectl get pods -n nim
```
And check that the pods have become "Ready".

Once that is true, you can try something like:

```bash
helm -n nim test my-nim --logs
```

Which will run some simple inference requests. If the three tests pass, you'll know the deployment was successful.

> Setting up external ingress without adding an authentication layer is not recommended because NIM doesn't provide authentication on its own. The chart provides options for basic ingress, if desired.

Reviewer: ```suggestion
Avoid setting up external ingress without adding an authentication layer. This is because NIM doesn't provide authentication on its own. The chart provides options for basic ingress.
```
